test/augmentation/test_augmentation.py

```diff
@@ -25,6 +25,334 @@
+
+    @pytest.mark.parametrize("input_shape", [(4, 5), (3, 4, 5), (2, 3, 4, 5)])
+    def test_consistent_output_shape(self, device, dtype, input_shape):
+        self._test_consistent_output_shape_implementation(
+            device=device, dtype=dtype, input_shape=input_shape, **self._default_param_set)
+
+    def test_random_p_0(self, device, dtype):
+        self._test_random_p_0_implementation(device=device, dtype=dtype, **self._default_param_set)
+
+    def test_random_p_0_return_transform(self, device, dtype):
+        self._test_random_p_0_return_transform_implementation(device=device, dtype=dtype, **self._default_param_set)
+
+    def test_random_p_1(self, device, dtype):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_random_p_1_return_transform(self, device, dtype):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_exception(self):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_batch(self):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_jit(self):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_gradcheck(self):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_sequential(self, device, dtype):
+        raise NotImplementedError("Implement a stupid routine.")
+
+    def test_random_batch_transform(self, device, dtype):
+        raise NotImplementedError("Implement a stupid routine.")
+
+# TODO Implement
+# test_random_p_half_return_transform (manual force [True, False] for batchsize 2 inputs)
+# test_batch
+# test_batch_return_transform
+# test_sequential
+# test_sequential_return_transform
+# test_coordinate check
+# test_jit
+# test_gradcheck
+
+    def _create_augmentation_from_params(self, **params):
+        return self._augmentation_cls(**params)
+
+    ############################################################################################################
+    # Test case implementations
+    ############################################################################################################
+
+    def _test_smoke_implementation(self, device, dtype, **params):
+        assert issubclass(self._augmentation_cls,
+                          AugmentationBase2D), f"{self._augmentation_cls} is not a subclass of AugmentationBase2D"
+
+        # Can be instatiated
+        augmentation = self._create_augmentation_from_params(**params, return_transform=False)
+        assert issubclass(
+            type(augmentation), AugmentationBase2D), f"{type(augmentation)} is not a subclass of AugmentationBase2D"
+
+        # generate_parameters can be called and returns the correct amount of parameters
+        batch_shape = (4, 3, 5, 6)
+        generated_params = augmentation.generate_parameters(batch_shape)
+        assert isinstance(generated_params, dict)
+        generated_params["batch_prob"] = torch.tensor([True, True, True, True])
```

Comment:
Just to mention that the probabilities generation has been changed to a routine to avoid unnessasary computations:
```
select_images => generate_param_for_selected_images => apply_transform => replace_the_original_images
```
Thus, you may simply do:
```
generated_params = augmentation.generate_parameters((4, *batch_shape[1:]))
```
Then
```
images[batch_prob] = augmentation(selected_images, generated_params)
```
if you want params for 4 images.